ecs_client_test.go

```diff
@@ -16,14 +16,39 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestECSClientInterface(t *testing.T) {
+func TestECSClient(t *testing.T) {
 	assert.Implements(t, (*ECSClient)(nil), &BasicECSClient{})
 }
+func TestECSClientTaskDefinition(t *testing.T) {
+
+	cleanupTaskDefinition := func(ctx context.Context, t *testing.T, c *BasicECSClient, out *ecs.RegisterTaskDefinitionOutput) {
+		if out != nil && out.TaskDefinition != nil && out.TaskDefinition.TaskDefinitionArn != nil {
+			out, err := c.DeregisterTaskDefinition(ctx, &ecs.DeregisterTaskDefinitionInput{
+				TaskDefinition: out.TaskDefinition.TaskDefinitionArn,
+			})
+			require.NoError(t, err)
+			require.NotZero(t, out)
+		}
+	}
+
+	cleanupTask := func(ctx context.Context, t *testing.T, c *BasicECSClient, runOut *ecs.RunTaskOutput) {
+		if runOut != nil && len(runOut.Tasks) > 0 && runOut.Tasks[0].TaskArn != nil {
+			out, err := c.StopTask(ctx, &ecs.StopTaskInput{
+				Cluster: aws.String(os.Getenv("AWS_ECS_CLUSTER")),
+				Task:    aws.String(*runOut.Tasks[0].TaskArn),
+			})
+			require.NoError(t, err)
+			require.NotZero(t, out)
+		}
+	}
+
+	makeFamily := func() string {
+		return os.Getenv("AWS_ECS_TASK_DEFINITION_PREFIX") + utility.RandomString()
```

Comment:
This should also take a name (i.e. `t.Name()` like for the Secrets Manager test) and use `path.Join` so we're consistent with the Secrets Manager tests.

Reply:
`Family` doesn't allow backslash characters, so I'm going to join the strings with hyphens manually. Does that sound okay?

Reply:
Sure, that works. You can probably just copy-paste `makeTestSecret` and replace `path.Join` with `strings.Join`.